src/index.js

```diff
@@ -457,6 +545,60 @@ class Searchbase {
     this.setValue(value);
   };
 
+  //mic event
+  onMicClick = () => {
+    const prevStatus = this._micStatus;
+    if (window.SpeechRecognition && prevStatus !== MIC_STATUS.denied) {
+      if (prevStatus === MIC_STATUS.active) {
+        this._micStatus = MIC_STATUS.inactive;
+        this._applyOptions(
```

Comment:
Create a method `_setMicStatus` to set the mic status and apply the options.
You're repeating the code below at line no 572.